computation-slurm/src/main/java/com/powsybl/computation/slurm/SlurmComputationManager.java

```diff
@@ -215,26 +217,47 @@ public OutputStream newCommonFile(String fileName) throws IOException {
                 LOGGER.error(e.toString(), e);
                 f.completeExceptionally(e);
             }
-            taskStore.remove(f);
+            if (!f.isCancelled()) {
+                taskStore.clearBy(f);
+            }
         });
         return f;
     }
 
-    static class Mycf<R> extends CompletableFuture<R> {
+    static class SlurmCompletableFuture<R> extends CompletableFuture<R> {
 
-        Thread thread;
-        SlurmComputationManager mgr;
-        volatile boolean cancel = false;
+        private Thread thread;
+        private SlurmComputationManager mgr;
+        private volatile boolean cancel = false;
+        private volatile boolean cancelledBySlurm = false;
 
-        Mycf(SlurmComputationManager manager) {
-            mgr = manager;
+        private SlurmException exception;
+
+        SlurmCompletableFuture(SlurmComputationManager manager) {
+            mgr = Objects.requireNonNull(manager);
         }
 
         @Override
         public boolean isCancelled() {
             return cancel;
         }
 
+        // called by monitors, it differs from cancel() by client-side
+        boolean cancelBySlurm(SlurmException exception) {
+            Objects.requireNonNull(exception);
+            this.exception = exception;
+            cancelledBySlurm = true;
+            return cancel(true);
+        }
+
+        SlurmException getException() {
+            return exception;
+        }
+
+        private boolean isCancelledBySlurm() {
+            return cancelledBySlurm;
+        }
+
         @Override
         public boolean cancel(boolean mayInterruptIfRunning) {
```

Comment:
What about using the standard `CompletableFuture` with a handling of `CancellationException`, as proposed in [this PR](https://github.com/powsybl/powsybl-core/pull/859/files) for local computation manager ?

Reply:
As discussed just now : let's postpone this possible refactoring to after we have some feedback on the PR linked above.